core/templates/dev/head/components/gadget_panel.html

```diff
@@ -1,6 +1,6 @@
 <script type="text/ng-template" id="components/gadgetPanel">
-  <div ng-repeat="gadgetSpec in panelContents()">
-    <oppia-gadget class="protractor-test-<[gadgetSpec.gadget_type]>-gadget" gadget-type="gadgetSpec.gadget_type" gadget-customization-args="gadgetSpec.customization_args" show-in-states="gadgetSpec.visible_in_states">
+  <div ng-repeat="gadgetInstance in panelContents()">
+    <oppia-gadget class="protractor-test-<[gadgetInstance.gadget_type]>-gadget" gadget-type="gadgetInstance.gadget_type" gadget-name="gadgetInstance.gadget_name" gadget-customization-args="gadgetInstance.customization_args" show-in-states="gadgetInstance.visible_in_states">
```

Comment:
Ah, good point.

I think, long term, we can do something like adding a custom protractor-gadget-name attribute to the tag. But that introduces a new pattern, so let's not do it now.

Instead, in the interim, I'd just add a note along these lines:
```
   <!--
     TODO(anuzis): it's possible for multiple gadgets in an exploration to have the
     same type. In such cases, the protractor class used here does not uniquely
     identify the gadget. Find a way to fix this.
   -->
```
and leave it at that for now.